lib/std/os/windows/kernel32.zig

```diff
@@ -115,340 +78,559 @@ pub extern "kernel32" fn CreateNamedPipeW(
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn DeviceIoControl(
-    h: HANDLE,
-    dwIoControlCode: DWORD,
-    lpInBuffer: ?*const anyopaque,
-    nInBufferSize: DWORD,
-    lpOutBuffer: ?LPVOID,
-    nOutBufferSize: DWORD,
-    lpBytesReturned: ?*DWORD,
-    lpOverlapped: ?*OVERLAPPED,
+// TODO(stephen): Wrapper around NtSetInformationFile + `FILE_POSITION_INFORMATION`.
+//  `FILE_STANDARD_INFORMATION` is also used if dwMoveMethod is `FILE_END`
+pub extern "kernel32" fn SetFilePointerEx(
+    hFile: HANDLE,
+    liDistanceToMove: LARGE_INTEGER,
+    lpNewFilePointer: ?*LARGE_INTEGER,
+    dwMoveMethod: DWORD,
 ) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn DeleteFileW(lpFileName: [*:0]const u16) callconv(WINAPI) BOOL;
+// TODO(stephen): Wrapper around NtSetInformationFile + `FILE_BASIC_INFORMATION`
+pub extern "kernel32" fn SetFileTime(
+    hFile: HANDLE,
+    lpCreationTime: ?*const FILETIME,
+    lpLastAccessTime: ?*const FILETIME,
+    lpLastWriteTime: ?*const FILETIME,
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn DuplicateHandle(hSourceProcessHandle: HANDLE, hSourceHandle: HANDLE, hTargetProcessHandle: HANDLE, lpTargetHandle: *HANDLE, dwDesiredAccess: DWORD, bInheritHandle: BOOL, dwOptions: DWORD) callconv(WINAPI) BOOL;
+pub extern "kernel32" fn WriteFile(
+    in_hFile: HANDLE,
+    in_lpBuffer: [*]const u8,
+    in_nNumberOfBytesToWrite: DWORD,
+    out_lpNumberOfBytesWritten: ?*DWORD,
+    in_out_lpOverlapped: ?*OVERLAPPED,
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn ExitProcess(exit_code: UINT) callconv(WINAPI) noreturn;
+// TODO(stephen): wrapper for NtQueryInformationFile + `FILE_STANDARD_INFORMATION`
+pub extern "kernel32" fn GetFileSizeEx(
+    hFile: HANDLE,
+    lpFileSize: *LARGE_INTEGER,
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn FindFirstFileW(lpFileName: [*:0]const u16, lpFindFileData: *WIN32_FIND_DATAW) callconv(WINAPI) HANDLE;
-pub extern "kernel32" fn FindClose(hFindFile: HANDLE) callconv(WINAPI) BOOL;
-pub extern "kernel32" fn FindNextFileW(hFindFile: HANDLE, lpFindFileData: *WIN32_FIND_DATAW) callconv(WINAPI) BOOL;
+// TODO(stephen): Wrapper around GetStdHandle + NtFlushBuffersFile.
+pub extern "kernel32" fn FlushFileBuffers(
+    hFile: HANDLE,
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn FormatMessageW(dwFlags: DWORD, lpSource: ?LPVOID, dwMessageId: Win32Error, dwLanguageId: DWORD, lpBuffer: [*]u16, nSize: DWORD, Arguments: ?*va_list) callconv(WINAPI) DWORD;
+// TODO(stephen): Wrapper around NtSetInformationFile + `FILE_IO_COMPLETION_NOTIFICATION_INFORMATION`.
+pub extern "kernel32" fn SetFileCompletionNotificationModes(
+    FileHandle: HANDLE,
+    Flags: UCHAR,
+) callconv(WINAPI) BOOL;
 
-pub extern "kernel32" fn FreeEnvironmentStringsW(penv: [*:0]u16) callconv(WINAPI) BOOL;
+// TODO(stephen): `RtlGetCurrentDirectory_U(nBufferLength * 2, lpBuffer)`
+pub extern "kernel32" fn GetCurrentDirectoryW(
+    nBufferLength: DWORD,
+    lpBuffer: ?LPWSTR,
```

Comment:
I'm not sure that `?LPWSTR` here is an improvement over `?[*]WCHAR`, since this parameter does not need to have a sentinel (however, after the function is called, a sentinel *will* be written to the buffer).

To give some reasoning, the updated usage of `GetCurrentDirectoryW` looks like this:

```zig
    var wtf16le_buf: [PATH_MAX_WIDE:0]u16 = undefined;
    const result = kernel32.GetCurrentDirectoryW(wtf16le_buf.len, &wtf16le_buf);
```

which is subtly incorrect, since the length given to `GetCurrentDirectoryW` does not include the sentinel. That is, without passing `wtf16le_buf.len + 1` the buffer having a sentinel is irrelevant, and it's not fully intuitive to pass `len + 1`.

Reply:
Sounds reasonable, I just copied over the definition from the `processenv.h` header.